src/jobservice/job/impl/gc/garbage_collection.go

```diff
@@ -240,3 +246,54 @@ func (gc *GarbageCollector) deleteCandidates(ctx job.Context) error {
 	flushTrash = true
 	return nil
 }
+
+// clean the untagged blobs in each project, these blobs are not referenced by any manifest and will be cleaned by GC
+func (gc *GarbageCollector) removeUntaggedBlobs(ctx job.Context) {
+	// get all projects
+	projects := func(chunkSize int) <-chan *models.Project {
+		ch := make(chan *models.Project, chunkSize)
+
+		go func() {
+			defer close(ch)
+
+			params := &models.ProjectQueryParam{
+				Pagination: &models.Pagination{Page: 1, Size: int64(chunkSize)},
+			}
+
+			for {
+				results, err := project.Ctl.List(ctx.SystemContext(), params, project.Metadata(false))
+				if err != nil {
+					gc.logger.Errorf("list projects failed, error: %v", err)
+					return
+				}
+
+				for _, p := range results {
+					ch <- p
+				}
+
+				if len(results) < chunkSize {
+					break
+				}
+
+				params.Pagination.Page++
+			}
+
+		}()
+
+		return ch
+	}(50)
+
+	for project := range projects {
+		all, err := gc.blobMgr.List(ctx.SystemContext(), blob.ListParams{
```

Comment:
This may have performance problem when there are many many blobs in the project.

Reply:
Is it possible to introduce a prune method in blob ctl/mgr to do the same work?

Reply:
@heww compares with the performance of GC, it can be ignored?

Reply:
@ywk253100 you mean combind these two methods?